prov/gni/test/api_cq.c

```diff
@@ -291,6 +291,43 @@ void api_cq_recv_allowed(ssize_t sz, uint64_t flags, char *fn)
 TestSuite(api_cq, .init = api_cq_setup, .fini = api_cq_teardown,
 	  .disabled = false);
 
+void api_cq_wait1(struct fid_cq *cq0, uint64_t cq_bind_flags)
+{
+	int ret;
+	struct fi_cq_tagged_entry cqe;
+
+	if (!cq_bind_flags)
+		return;
+
+	while ((ret = fi_cq_read(msg_cq[0], &cqe, 1)) == -FI_EAGAIN);
```

Comment:
fi_cq_read can return with a -errno that is not -FI_EAGAIN. Do you want to handle that error case because if you don't handle it the test will move on even though an error occured.